### PR TITLE
updates for WildFly 14

### PR DESCRIPTION
**Pull request by Ladicek**
Updates needed for WildFly 14. This PR is not to be merged yet -- even the code quality is somewhat sub-par in certain areas. Reviews are welcome though!

### CODE REVIEW

**Comment by Ladicek** [comment]
I've done a bunch of changes to this PR since originally submitting it, and I'd like to consider merging it. I'd welcome any comments around the `HackedModuleDescriptor` stuff, the rest of the PR should be good.

**Comment by sberyozkin** [comment]
LGTM, can approve if no more comments will come.

File: src/main/java/org/wildfly/swarm/plugin/process/CDIMarker.java
Comment (michalszynkiewicz) [inline src/main/java/org/wildfly/swarm/plugin/process/CDIMarker.java:37]: What do we gain by being more restrictive here and in the other places where `log.error` was translated to throwing an exception?

Comment (Ladicek) [reply src/main/java/org/wildfly/swarm/plugin/process/CDIMarker.java:37]: Failing a build when the necessary pieces of a fraction are not generated successfully. The fraction just isn't complete.

File: src/main/java/org/wildfly/swarm/plugin/utils/HackedModuleDescriptor.java
Comment (michalszynkiewicz) [inline src/main/java/org/wildfly/swarm/plugin/utils/HackedModuleDescriptor.java]: Since it's for required for writing only maybe `ReadWriteModuleDescriptor` would be a better name?
Or `NamespacePreservingModuleDescriptor`?

File: src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java
Comment (michalszynkiewicz) [inline src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java:91]: why do we need to exclude platform modules now and didn't have to do it earlier?

Comment (Ladicek) [reply src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java:91]: Because there were no platform modules previously :-) JBoss Modules is JPMS-ready (well, sort of, I don't really know) and emulates the platform modules on Java 8. In any case (be it Java 8 or Java 9+), there is no `module.xml` for these modules that could be analyzed.

File: src/main/java/org/wildfly/swarm/plugin/utils/HackedModuleDescriptor.java
Comment (Ladicek) [reply src/main/java/org/wildfly/swarm/plugin/utils/HackedModuleDescriptor.java]: I like `NamespacePreservingModuleDescriptor`!

**Comment by michalszynkiewicz** [comment]
The topic still says [DO NOT MERGE], is still true?

**Comment by Ladicek** [comment]
I just pushed the changes to rename the `ModuleDescriptor` and removed "[DO NOT MERGE]" tag.

**Review by michalszynkiewicz — APPROVED** [review]

**Comment by Ladicek** [comment]
Added 2 more commits.

File: src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java
Comment (sberyozkin) [inline src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java]: This will log "totalSize: 0+ MB" in case of the unknown size. Maybe "totalSize: unknown" ? '0+' is a bit cryptic unless it is a convention I'm unaware of...

Comment (Ladicek) [reply src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java]: Yea, if no artifact could be resolved, then it would indeed print `0+ MB`, which looks weird. Typically, some artifacts could be resolved and some couldn't, in which case it would print e.g. `2.35+ MB`, which I think is a well-known convention for "more than 2.35 MB".

But I agree that for the zero case, it should indeed print "unknown size".

Comment (Ladicek) [reply src/main/java/org/wildfly/swarm/plugin/process/ModuleFiller.java]: Fixed.

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
@Ladicek I somehow lost my earlier comment about renaming "remove-artifact" to "remove" given that other directives also imply that (may be module even). Np if you prefer to keep "remove-artifact", all LGTM

**Comment by Ladicek** [comment]
The clause names are not very descriptive, but all refer to module dependencies AFAIK. This new one refers to artifacts, so I favor a more explicit name. Thanks for review!

**Pull request merged**
